packages/hash/api/src/model/knowledge/entity.model.ts

```diff
@@ -1,34 +1,38 @@
 import { ApolloError } from "apollo-server-errors";
 import {
-  Entity,
   GraphApi,
-  Subgraph,
   EntityStructuralQuery,
   Filter,
-  EntityMetadata,
 } from "@hashintel/hash-graph-client";
-
+import { Entity, Subgraph } from "@hashintel/hash-subgraph";
+import { getRootsAsEntities } from "@hashintel/hash-subgraph/src/stdlib/element/entity";
 import {
   EntityModel,
   EntityTypeModel,
-  LinkModel,
+  LinkEntityModel,
   LinkModelCreateParams,
-  LinkTypeModel,
 } from "..";
 import {
   PersistedEntityDefinition,
   PersistedLinkedEntityDefinition,
 } from "../../graphql/apiTypes.gen";
 import { exactlyOne, linkedTreeFlatten } from "../../util";
 
+/**
+ * @todo: import this directly from `@hashintel/hash-subgraph` once it is exported
```

Comment:
Just commenting here for visibility, upstream PR was merged and we can address this now after updating branch

Reply:
Something I forgot to mention is that there is no export of `EntityProperties` or `JsonObject`. Is that something we want to do? If not I'll remove the `@todo` entirely here.

Reply:
Partially addressed in https://github.com/hashintel/hash/pull/1435/commits/b989e43090dec04e6da28fbe200c608b7fe7ad83

Reply:
I missed that you're calling it `PropertyObject, have addressed in https://github.com/hashintel/hash/pull/1435/commits/1190134e21615fd3eb5b21ee7a845e044995ad39